### PR TITLE
Generate categories

### PR DESCRIPTION
**Pull request by Hofer-Julian**

### CODE REVIEW

File: make_docs.nu [hunk 1]
```diff
@@ -136,3 +144,51 @@ $"($example.description)
   )
   $doc
 }
+
+def generate-category-sidebar [unique_categories] {
+  let sidebar_path = (['.', '.vuepress', 'configs', "sidebar", "command_categories.ts"] | path join)
+  let list_content = (
+    $unique_categories |
+    each { |category| $category | str replace '\?' '' | str replace ' ' '_' } |
```
Comment (hustcer) [inline make_docs.nu:152]: `str replace -a ' ' '_'`

File: make_docs.nu [hunk 2]
```diff
@@ -136,3 +144,51 @@ $"($example.description)
...
+  )
+  $"export const commandCategories = [
+($list_content)
+];
+" | save --raw --force $sidebar_path
+}
+
+
+def generate-category [category] {
+  let safe_name = ($category | str replace '\?' '' | str replace ' ' '_')
```
Comment (hustcer) [inline make_docs.nu:164]: `str replace -a ' ' '_'`

File: make_docs.nu [hunk 3]
```diff
@@ -136,3 +144,51 @@ $"($example.description)
...
+  let doc_path = (['.', 'commands', 'categories', $'($safe_name).md'] | path join)
+  $"# ($category)
```
Comment (hustcer) [inline make_docs.nu:166]: `($category | str title-case)` will make docs look better

File: make_docs.nu (hunk 2 above)
Comment (Hofer-Julian) [reply make_docs.nu]: Done, and extracted into `safe-path`

File: make_docs.nu (hunk 3 above)
Comment (Hofer-Julian) [reply make_docs.nu]: Nice, I was looking for something like this, but missed it.

**Comment by Hofer-Julian** [comment]
> Can we also fix the build so that we get the Windows-only commands like `registry query`. I think that's the only one but I'm not positive.

I am not sure how I would do this.

**Comment by fdncred** [comment]
> I am not sure how I would do this.

probably not for this PR. @hustcer we need to include the windows tools as well.

The problem is, that if we build and generate only on mac or linux, we don't get the windows tool(s).

**Comment by hustcer** [comment]
> probably not for this PR. @hustcer we need to include the windows tools as well.
> The problem is, that if we build and generate only on mac or linux, we don't get the windows tool(s).

I see, I will try to add that later

**Comment by hustcer** [comment]
@Hofer-Julian `commands/categories/dataframe_or lazyframe.md` I think we can remove this file.

**Comment by Hofer-Julian** [comment]
> @Hofer-Julian `commands/categories/dataframe_or lazyframe.md` I think we can remove this file.

Good catch, done.

**Comment by hustcer** [comment]
Thanks, let's merge it.

**Pull request merged**